DeepSpeech.py

```diff
@@ -3,179 +3,168 @@
 
 # Step widths
 
-# The number of epochs we cycle through before displaying progress
-display_step = int(os.environ.get('ds_display_step', 1))
-
-# The number of epochs we cycle through before checkpointing the model
-checkpoint_step = int(os.environ.get('ds_checkpoint_step', 5))
-
-# The number of epochs we cycle through before validating the model
-validation_step = int(os.environ.get('ds_validation_step', 0))
-
+tf.app.flags.DEFINE_integer ('display_step',     1,           'number of epochs we cycle through before displaying progress')
+tf.app.flags.DEFINE_integer ('validation_step',  0,           'number of epochs we cycle through before validating the model')
 
 # Checkpointing
 
-# The directory in which checkpoints are stored
-checkpoint_dir = os.environ.get('ds_checkpoint_dir', xdg.save_data_path('deepspeech'))
+tf.app.flags.DEFINE_string  ('checkpoint_dir',   '',          'directory in which checkpoints are stored')
+tf.app.flags.DEFINE_integer ('checkpoint_secs',  600,         'checkpoint saving interval in seconds')
+
+# Exporting
 
-# Whether to resume from checkpoints when training
-restore_checkpoint = bool(int(os.environ.get('ds_restore_checkpoint', 0)))
+tf.app.flags.DEFINE_string  ('export_dir',       '',          'directory in which exported models are stored')
+tf.app.flags.DEFINE_integer ('export_version',   1,           'version number of the exported model')
+tf.app.flags.DEFINE_boolean ('remove_export',    False,       'weather to remove old exported models')
 
-# The directory in which exported models are stored
-export_dir = os.environ.get('ds_export_dir', None)
+# Reporting
 
-# The version number of the exported model
-export_version = 1
+tf.app.flags.DEFINE_integer ('log_level',        1,           'log level for console logs - 0: INFO, 1: WARN, 2: ERROR, 3: FATAL')
+tf.app.flags.DEFINE_boolean ('log_traffic',      False,       'log cluster transaction and traffic information during debug logging')
 
-# Whether to remove old exported models
-remove_export = bool(int(os.environ.get('ds_remove_export', 0)))
+tf.app.flags.DEFINE_boolean ('publish_wer_log',  False,       'weather to publish the WER log')
+tf.app.flags.DEFINE_string  ('wer_log_file',     'werlog.js', 'log-file for keeping track of WER progress')
 
+tf.app.flags.DEFINE_boolean ('log_placement',    False,       'weather to log device placement of the operators to the console')
+tf.app.flags.DEFINE_integer ('report_count',     10,          'number of phrases to print out during a WER report')
 
-# Reporting
+tf.app.flags.DEFINE_boolean ('log_variables',    False,       'weather to log gradients and variables summaries to TensorBoard during training')
+tf.app.flags.DEFINE_integer ('summaries_steps',  1,           'number of global training steps we cycle through before saving a summary')
+tf.app.flags.DEFINE_string  ('logs_dir',         'logs',      'directory in which checkpoints are stored')
```

Comment:
Removed/Renamed it to new "summary" FLAGS that also default to XDG directory logic like checkpoints do. Look for "summary" in the DeepSpeech code to re-review.